File: kafka-helper-domain/src/main/java/com/hg/kafka/helper/domain/repository/HashkeyInfoRepository.java
package com.hg.kafka.helper.domain.repository;

import com.hg.kafka.helper.domain.entity.HashkeyInfo;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface HashkeyInfoRepository extends MongoRepository<HashkeyInfo, String> {
    List<HashkeyInfo> findAllByTopic(String topic);
    Optional<HashkeyInfo> findOneByTopicAndHashkey(String topic, String hashkey);
    Optional<HashkeyInfo> findOneByGroupId(String groupId);
}

File: kafka-helper-domain/src/main/kotlin/com/hg/kafka/helper/domain/repository/HashkeyInfoRepository.kt
package com.hg.kafka.helper.domain.repository

import com.hg.kafka.helper.domain.entity.HashkeyInfo
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface HashkeyInfoRepository : MongoRepository<HashkeyInfo, String> {
    fun findAllByTopic(topic: String): List<HashkeyInfo>
    fun findOneByTopicAndHashkey(topic: String, hashkey: String): Optional<HashkeyInfo>
    fun findOneByGroupId(groupId: String): Optional<HashkeyInfo>
}